Client/Frontend/Library/LibraryPanelHelper.swift

```diff
@@ -22,6 +14,42 @@ protocol LibraryPanelDelegate: AnyObject {
     func libraryPanel(didSelectURLString url: String, visitType: VisitType)
 }
 
+protocol LibraryPanel: UIViewController, NotificationThemeable {
+    var libraryPanelDelegate: LibraryPanelDelegate? { get set }
+    var state: LibraryPanelMainState { get set }
+    var bottomToolbarItems: [UIBarButtonItem] { get }
+
+    func handleLeftTopButton()
+    func handleRightTopButton()
+    func shouldDismissOnDone() -> Bool
+}
+
+extension LibraryPanel {
+    var flexibleSpace: UIBarButtonItem {
+        return UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+    }
+
+    func updatePanelState(newState: LibraryPanelMainState) {
+        state = newState
+    }
+
+    func shouldDismissOnDone() -> Bool {
+        return true
+    }
+
+    func handleLeftTopButton() {
+        // no implementation needed
```

Comment:
No implementation needed since it's handled elsewhere? Just wanna make sure I understand the comment there

Reply:
because `ReaderPanel` and `DownloadsPanel` don't have any Left top button